tsflex/features/segmenter/strided_rolling.py

```diff
@@ -250,20 +250,69 @@ def apply_func(self, func: FuncWrapper) -> pd.DataFrame:
         # expression only once, whereas a list comprehension evaluates its expression
         # every time).
         # See more why: https://stackoverflow.com/a/59838723
-        out = np.array(
-            list(
-                map(
-                    func,
-                    *[
-                        [
-                            sc.values[sc.start_indexes[idx]: sc.end_indexes[idx]]
-                            for idx in range(len(self.index))
-                        ]
-                        for sc in self.series_containers
-                    ],
+        out: np.array = None
+        if func.vectorized:
+            # Vectorized function execution
+
+            ## IMPL 1
+            # out = np.asarray(
+            #         func(
+            #             *[
+            #                 np.array([
+            #                     sc.values[sc.start_indexes[idx]: sc.end_indexes[idx]]
+            #                     for idx in range(len(self.index))
+            #                 ])
+            #                 for sc in self.series_containers
+            #             ],
+            #         )
+            #     )
+
+            ## IMPL 2
+            # out = np.asarray(
```

Comment:
""

Reply:
Is a good implementation imo (equivalent to the one below) -> but does not make any of the required assertions